src/mongo/util/net/listen.cpp

```diff
@@ -51,6 +52,23 @@
 
 #endif
 
+// Make sure our Linux has the timer we need
+#if defined(__linux__)
+    #ifndef CLOCK_MONOTONIC_COARSE
+        #error "Your ancient kernel doesn't support CLOCK_MONOTONIC_COARSE: upgrade to v2.6.31 or newer"
```

Comment:
Now that it's almost two years later, I'm wondering are you still targeting the same kernel versions? I'd love to see this PR make it's way in so that Mongo stops killing my laptop battery